src/utils/bars.ts

```diff
@@ -0,0 +1,34 @@
+import { ScaleBand } from 'd3-scale';
+import get from 'lodash.get';
+
+import { EGroupedBarLayout, IHistogramBar } from '../Histogram';
+
+export const groupedBarsUseSameXAxisValue = ({ groupLayout, stacked }): boolean => {
+    return stacked || groupLayout === EGroupedBarLayout.STACKED || groupLayout === EGroupedBarLayout.OVERLAID;
+};
+
+export const barMargin = (bar: IHistogramBar): number => {
+    const m = get(bar, 'margin', 0);
+    return m >= 0 && m <= 1
+        ? m
+        : 0.1;
+};
+
+export const getBarWidth = (
+    i: number,
+    groupLayout: EGroupedBarLayout,
+    bar: IHistogramBar,
+    innerScaleBand: ScaleBand<string>,
+) => {
```

Comment:
is it worth having a return type here? although I'm guessing it's getting inferred from line 26

Reply:
yeah its inferred so no need